src/datamodel.js

```diff
@@ -446,10 +446,14 @@ class DataModel extends Relation {
         const fs = clone.getFieldspace().fields;
         const suppliedFields = depFieldIndices.map(idx => fs[idx]);
 
+        let cachedClonedDm;
+        let cachedStore = {};
+        let cloneProvider = () => { if (!cachedClonedDm) cachedClonedDm = this.detachedRoot(); return cachedClonedDm; };
```

Comment:
`cloneProvider` won't cache the datamodel instance. It's up to the user to cache it.

Reply:
Okay, then we need to provide the best practice to the user to use properly as the clone might occur multiple times as follows:

```javascript

const selectedDm = dm.select((fields, i, cloneProvider, store) => {
	if(!store.cachedDm) {
		// need to cache it
		store.cachedDm = cloneProvider();
	}

	// blah blah
});

```